Resources/Catalog/FileCatalog.py

```diff
@@ -300,50 +314,53 @@ def removeCatalog( self, catalogName ):
 
   def _getSelectedCatalogs( self, desiredCatalogs ):
     for catalogName in desiredCatalogs:
-      res = self._getCatalogConfigDetails( catalogName )
-      if not res['OK']:
-        return res
-      catalogConfig = res['Value']
-      res = self._generateCatalogObject( catalogName )
-      if not res['OK']:
-        return res
-      oCatalog = res['Value']
-      self.readCatalogs.append( ( catalogName, oCatalog, True ) )
-      self.writeCatalogs.append( ( catalogName, oCatalog, True ) )
+      result = self._getCatalogConfigDetails( catalogName )
+      if not result['OK']:
+        return result
+      catalogConfig = result['Value']
+      result = self._generateCatalogObject( catalogName )
+      if not result['OK']:
+        return result
+      oCatalog = result['Value']
+      self.readCatalogs.append( ( catalogName, oCatalog, catalogConfig['Master'] ) )
+      self.writeCatalogs.append( ( catalogName, oCatalog, catalogConfig['Master'] ) )
     return S_OK()
 
-  def _getCatalogs( self ):
+  def _getEligibleCatalogs( self ):
```

Comment:
I must confess that I am myself a bit confused about which part of the configuration should go to resource, which part should go to operation. is there a doc about this ? The only one I found is this http://diracgrid.org/files/docs/AdministratorGuide/Configuration/ConfReference/Resources/FileCatalogs/index.html